src/test/java/cloud/eppo/FlagEvaluatorTest.java

```diff
@@ -0,0 +1,427 @@
+package cloud.eppo;
+
+import static cloud.eppo.Utils.base64Encode;
+import static cloud.eppo.Utils.getMD5Hex;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import cloud.eppo.model.ShardRange;
+import cloud.eppo.ufc.dto.Allocation;
+import cloud.eppo.ufc.dto.EppoValue;
+import cloud.eppo.ufc.dto.FlagConfig;
+import cloud.eppo.ufc.dto.OperatorType;
+import cloud.eppo.ufc.dto.Shard;
+import cloud.eppo.ufc.dto.Split;
+import cloud.eppo.ufc.dto.SubjectAttributes;
+import cloud.eppo.ufc.dto.TargetingCondition;
+import cloud.eppo.ufc.dto.TargetingRule;
+import cloud.eppo.ufc.dto.Variation;
+import cloud.eppo.ufc.dto.VariationType;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.Test;
+
+public class FlagEvaluatorTest {
```

Comment:
Mostly not new code; largely ported from Android implementation ([source](https://github.com/Eppo-exp/android-sdk/blob/main/eppo/src/test/java/cloud/eppo/android/FlagEvaluatorTest.java))